compiler/rustc_codegen_llvm/src/type_.rs

```diff
@@ -44,11 +44,13 @@ impl<'ll> CodegenCx<'ll, '_> {
     pub(crate) fn set_struct_body(&self, ty: &'ll Type, els: &[&'ll Type], packed: bool) {
         unsafe { llvm::LLVMStructSetBody(ty, els.as_ptr(), els.len() as c_uint, packed as Bool) }
     }
-
+}
+impl<'ll> SimpleCx<'ll> {
     pub(crate) fn type_void(&self) -> &'ll Type {
         unsafe { llvm::LLVMVoidTypeInContext(self.llcx) }
     }
-
+}
+impl<'ll, 'tcx> CodegenCx<'ll, 'tcx> {
```

Comment:
I think you can be much more aggressive at moving things into `SimpleCx` Basically just change all inherent impls and only move those functions back to `CodegenCx` that need it.

but it's also fine to do this in follow-up PRs when you actually need stuff

Reply:
I moved over all the basic impl, but not the trait implementations, since I'm not sure if I could implement all of them with the SimpleCx (probably not).